test/v2/core/sgETH.spec.js

```diff
@@ -0,0 +1,56 @@
+const {ethers} = require("hardhat");
+const {expect} = require("chai");
+
+describe("SgETH.sol", () => {
+  let sgEth, deployer, alice;
+  let MINTER_ROLE;
+
+  beforeEach(async () => {
+    const [owner, addr1] = await ethers.getSigners();
+
+    const SgETH = await ethers.getContractFactory("SgETH");
+    sgEth = await SgETH.deploy([]);
+    await sgEth.deployed();
+
+    deployer = owner;
+    alice = addr1;
+
+    MINTER_ROLE = await sgEth.MINTER();
+  });
+
+  it("addMinter", async () => {
```

Comment:
can we add a e2e test? 
The expected use case is in deploy/v2/deploy_minterv2_follow.js  
e2e would look like:
1. deploy sgeth 
2. add secondary minter contract / eoa 
3. revoke deployer minter rights 
4. add secondary owner 
5. revoke deployer admin rights 
6. check auth invariants are preserved. i.e ex owner and outsiders cannot interact with the contract

Reply:
what's the secondary minter contract?

Reply:
this one 
https://github.com/chimera-defi/SharedDeposit/blob/main/contracts/v2/core/SharedDepositMinterV2.sol

Reply:
what's the meaning of `revoke deployer minter rights`?